src/odemis/acq/stream/_helper.py

```diff
@@ -149,6 +168,175 @@ def __init__(self, name, detector, dataflow, emitter, scanner=None, sstage=None,
         except AttributeError:
             pass
 
+    # overrides method in _live.py
+    def _onNewData(self, dataflow, data):
+        """
+        Called when a new image has arrived from the detector.
+        Usually the dataflow subscribes to onNewData.
+        :param dataflow: (model.Dataflow) The dataflow.
+        :param data: (model.DataArray). The new image that has arrived from the detector.
+        """
+        if hasattr(self, "integrationTime"):
+            self.raw = (self.raw + [data])[-self.integrationCounts.value:]
+        else:
+            self.raw = (self.raw + [data])
+
+        # TODO both methods call integrateImage -> make easier?
```

Comment:
Yes, but not at the same frequency. I'd just leave it as-is. If the integration count is very large, the main issue might be memory, before the CPU (as, anyway, the new data wouldn't arrive fast).